nav2_behavior_tree/plugins/decorator/goal_updated_controller.cpp

```diff
@@ -37,18 +36,18 @@ BT::NodeStatus GoalUpdatedController::tick()
     // Reset since we're starting a new iteration of
     // the goal updated controller (moving from IDLE to RUNNING)
 
-    BT::getInputOrBlackboard("goals", goals_);
-    BT::getInputOrBlackboard("goal", goal_);
+    goals_ = config().blackboard->get<std::vector<geometry_msgs::msg::PoseStamped>>("goals");
```

Comment:
I don't know why, but `BT::getInputOrBlackboard("goals", goals_);` is not getting the goals set in the test whereas this version does.